app/models/course/assessment.rb

```diff
@@ -3,7 +3,10 @@
 # An assessment is a collection of questions that can be asked.
 class Course::Assessment < ActiveRecord::Base
   acts_as_lesson_plan_item
+  has_one_folder
 
+  after_initialize :build_initial_folder, if: :new_record?
+  before_validation :assign_folder_attributes
```

Comment:
Rule of thumb for refactoring: if it is used in one place, leave it. If it is used in two places, consider. If it is used in three places, refactor.